From 8a0d2ec58961eda74fe36dccf0a82a17d73b0278 Mon Sep 17 00:00:00 2001
From: Paul Adenot <paul@paul.cx>
Date: Wed, 5 Oct 2022 14:54:33 +0200
Subject: [PATCH] Use c_char instead of i8 to fix build on Linux aarch64.

It feels like std::ffi::c_char would be more appropriate, but this isn't
available in the minimum rust version supported by Firefox (it was just added
in 1.64, current Firefox must build with 1.61).
---
 cubeb-backend/src/log.rs | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/third_party/rust/cubeb-backend/src/log.rs b/third_party/rust/cubeb-backend/src/log.rs
index 16580e5..c03d6b8 100644
--- a/third_party/rust/cubeb-backend/src/log.rs
+++ b/third_party/rust/cubeb-backend/src/log.rs
@@ -3,6 +3,8 @@
 // This program is made available under an ISC-style license.  See the
 // accompanying file LICENSE for details.
 
+use std::os::raw::c_char;
+
 /// Maximum length in bytes for a log message.
 /// Longer messages are silently truncated.  See `write_str`.
 const LOG_LIMIT: usize = 1024;
@@ -55,7 +57,7 @@ impl<const N: usize> std::fmt::Write for StaticCString<N> {
 /// Formats `$file:line: $msg\n` into an on-stack buffer of size `LOG_LIMIT`,
 /// then calls `log_callback` with a pointer to the formatted message.
 pub fn cubeb_log_internal_buf_fmt(
-    log_callback: unsafe extern "C" fn(*const i8, ...),
+    log_callback: unsafe extern "C" fn(*const c_char, ...),
     file: &str,
     line: u32,
     msg: std::fmt::Arguments,

